microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusConfigRootProvider.java

```diff
@@ -318,10 +318,22 @@ private void processConfigGroup(String extensionName, IJavaElement javaElement,
 					} else {
 						subKey = baseKey + "." + name;
 					}
-					final String defaultValue = configItemAnnotation == null ? ConfigItem.NO_DEFAULT
-							: getAnnotationMemberValue(configItemAnnotation, "defaultValue");
-
+					 
 					String fieldTypeName = getResolvedTypeName(field);
+					

```

Comment:
@xorye a thing I don't understand is that the default value is computed in `addItemMetadata`, why you manage this default value here?

Reply:
oops, I initially had this because the default that is managed for boolean and numbers in `addItemMetadata` was only working if the `defaultValue` was equal to `<<no default>>` and was not managed if `defaultValue == null`.

In my latest push, I changed `addItemMetadata` to check if `defaultValue == null` for boolean and numbers.

Reply:
It's a simple and perfect fix @xorye !